src/module/server-instance/ServerInstanceCtrl.ts

```diff
@@ -0,0 +1,81 @@
+import {deepGet} from "../../common/utils/Utils";
+import {ServerService} from "../../services/server/ServerService";
+import {IServerAddress} from "../../services/server/IServerAddress";
+import {IServer} from "../../services/server/IServer";
+import {IIntervalService} from "angular";
+import {InstanceMemoryChart} from "./MemoryChart";
+
+export class ServerInstanceCtrl {
+    static $inject: string[] = ["$interval", "serverService", "coord", "serverInstance"];
+
+    threadCount: number;
+    threadPeakCount: number;
+    threadDaemonCount: number;
+
+    directBufferPoolCount: number;
+    directBufferPoolMemoryUsed: number;
+
+    mappedBufferPoolCount: number;
+    mappedBufferPoolMemoryUsed: number;
+
+    nodeStats: any;
+    chart: InstanceMemoryChart;
+
+    constructor(private $interval: IIntervalService, private serverService: ServerService,
+                public coord: IServerAddress, private serverInstance: IServer) {
+        this.$interval(() => (this.refreshStats()), 500, 1);
+    }
+
+    refreshStats(): void {
+        let server: IServerAddress = this.serverInstance.address;
+        this.fetchThreadAndMemoryStats(server);
+        this.fetchAggregateNodeStats(server);
+    }
+
+    fetchThreadAndMemoryStats(address: IServerAddress): void {
+        this.serverService.getServerStats(address).then(response => {
+            // memory
+            let memory: any = response.memory["heap-memory-usage"];
+            let used: number = (memory.used / 1024) / 1024;
+            let max: number = (memory.max / 1024) / 1024;
+
+            // threading
+            let threading: any = response.threading;
+            this.threadCount = threading["thread-count"];
+            this.threadPeakCount = threading["peak-thread-count"];
+            this.threadDaemonCount = threading["daemon-thread-count"];
+
+            let directBufferPool: any = deepGet(response, "buffer-pool.name.direct");
+            let mappedBufferPool: any = deepGet(response, "buffer-pool.name.mapped");
+
+            this.directBufferPoolCount = directBufferPool.count;
+            this.directBufferPoolMemoryUsed = directBufferPool["memory-used"];
+
+            this.mappedBufferPoolCount = mappedBufferPool.count;
+            this.mappedBufferPoolMemoryUsed = mappedBufferPool["memory-used"];
+
+            if (this.chart) {
+                this.chart.destroy();
+            }
+            this.chart = new InstanceMemoryChart("#chart", used, max);
+        });
+    }
+
+    fetchAggregateNodeStats(address: IServerAddress): void {
+
+        this.serverService.getAggregateNodeStats(address).then(response => {
```

Comment:
What happens here if this.serverInstance is stopped? this is possible as ServerService.getServer() simply returns a new Server instance when it a server has been stopped [1] and we will get errors

[1] https://github.com/vblagoje/infinispan-management-console/blob/1e887087d963c11032fd8d3c29c01d78bcfdb817/src/services/server/ServerService.ts#L52